node/bft/src/helpers/timing.rs

```diff
@@ -0,0 +1,317 @@
+// Copyright (c) 2019-2025 Provable Inc.
+// This file is part of the snarkOS library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+    time::SystemTime,
+};
+
+use once_cell::sync::Lazy;
+use serde::{Deserialize, Serialize};
+
+/// Global storage for round-based event data
+static ROUND_EVENTS: Lazy<Arc<RwLock<HashMap<u64, RoundEvents>>>> = Lazy::new(|| Arc::new(RwLock::new(HashMap::new())));
+
+/// Global storage for subdag-based timing data (using lowest/highest rounds from subdag)
+static SUBDAG_TIMINGS: Lazy<Arc<RwLock<HashMap<(u64, u64), SubdagTimings>>>> =
+    Lazy::new(|| Arc::new(RwLock::new(HashMap::new())));
+
+/// Consensus stages that occur per round
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum ConsensusStage {
+    ProposalSeen,
+    ProposalCreated,
+    CertificateAdded,
+}
+
+/// Block processing stages that occur per subdag
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum SubdagStage {
+    SubdagProcessing,
+    CheckNextBlock,
+    AdvanceToNextBlock,
+}
+
+/// Timing event for a specific consensus stage
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TimingEvent {
+    pub round: u64,
+    pub timestamp: SystemTime,
+    pub event_type: String,
```

Comment:
```suggestion
    pub event_type: &'static str,
```
judging by the current use

Reply:
Can't auto-deserialize into a static reference:

```
error: lifetime may not live long enough
  --> node/bft/src/helpers/timing.rs:62:5
   |
59 | #[derive(Debug, Clone, Serialize, Deserialize)]
   |                                   ----------- lifetime `'de` defined here
...
62 |     pub proposal_seen: Vec<TimingEvent>,
   |     ^^^ requires that `'de` must outlive `'static`
```